packages/@apphosting/adapter-nextjs/src/overrides.ts

```diff
@@ -1,5 +1,79 @@
 import { AdapterMetadata, MiddlewareManifest } from "./interfaces.js";
-import { loadRouteManifest, writeRouteManifest, loadMiddlewareManifest } from "./utils.js";
+import {
+  loadRouteManifest,
+  writeRouteManifest,
+  loadMiddlewareManifest,
+  exists,
+  writeFile,
+} from "./utils.js";
+import { join } from "path";
+import { readFileSync } from "fs";
+
+export async function overrideNextConfig(projectRoot: string, nextConfigFileName: string) {
+  // Check if the file exists in the current working directory
+  const configPath = join(projectRoot, nextConfigFileName);
+
+  if (!(await exists(configPath))) {
+    console.log(`No Next.js config file found at ${configPath}`);
+    return;
+  }
+
+  // Determine the file extension
+  const fileExtension = nextConfigFileName.split(".").pop() || "js";
+  const originalConfigName = `next.config.original.${fileExtension}`;
+  const newConfigName = `next.config.${fileExtension}`;
+
+  // Rename the original config file
+  try {
+    const originalContent = readFileSync(configPath, "utf-8");
+    await writeFile(join(projectRoot, originalConfigName), originalContent);
+
+    // Create a new config file with the appropriate import
+    let importStatement;
+    if (fileExtension === "js" || fileExtension === "cjs") {
+      importStatement = `const originalConfig = require('./${originalConfigName}');`;
+    } else if (fileExtension === "mjs") {
+      importStatement = `import originalConfig from './${originalConfigName}';`;
+    } else if (fileExtension === "ts") {
+      importStatement = `import originalConfig from './${originalConfigName.replace(".ts", "")}';`;
+    } else {
+      throw new Error(`Unsupported file extension: ${fileExtension}`);
+    }
+
+    // Create the new config content with our overrides
+    const newConfigContent = `${importStatement}
```

Comment:
Let's add `// @ts-nocheck` so we don't fail the build if we violate any of their typescript rules